File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        prefix = []
        product = 1

        for num in nums:
            prefix.append(product)
            product *= num

        suffix = []
        product = 1

        for num in reversed(nums):
            suffix.append(product)
            product *= num
        suffix.reverse()
        return [suffix[i] * prefix[i] for i in range(len(nums))]

File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-0.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int index = 0;
        for (int i = 0; i < nums.length; i++)
        {
            if (i > 0 && nums[i -1] == nums[i]) continue;
            nums[index] = nums[i];
            index++;
        }
        return index;
    }
}